lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -166,6 +168,21 @@ def create(self, subsection, course_structure, course):
             self._compute_and_save_grade(subsection, course_structure, course)
         )
 
+    def update(self, student, usage_key, course_key):
+        """
+        Updates the SubsectionGrade object for the student and subsection.
+        """
+        from courseware.courses import get_course_by_id  # avoids circular import with courseware.py
+        course = get_course_by_id(course_key, depth=0)
```

Comment:
@nasthagiri I moved this code here, but it didn't resolve the circular dependency issue in either file, it just passed the buck as far as I can tell. Is there a way to avoid circular dependencies by having this code in this file, or does it just make the signal handler cleaner?